frontend/packages/kubevirt-plugin/integration-tests/tests/mocks/vmBuilderPresets.ts

```diff
@@ -0,0 +1,124 @@
+import { deepFreeze } from '../utils/utils';
+import { VMBuilder } from '../models/vmBuilder';
+import * as Combinatorics from 'js-combinatorics';
+import { Flavor, Workload, OperatingSystem, ProvisionSource } from '../utils/constants/wizard';
+import { VMBuilderDataGenerationConfig, VMBuilderData } from '../types/vm';
+import { testName } from '@console/internal-integration-tests/protractor.conf';
+import {
+  provisionSources,
+  rootDisk,
+  multusNetworkInterface,
+  containerRootDisk,
+  flavorConfigs,
+  getDiskToCloneFrom,
+} from './mocks';
+import { VirtualMachine } from '../models/virtualMachine';
+import { VMTemplateBuilder } from '../models/vmtemplateBuilder';
+import { VirtualMachineTemplate } from '../models/virtualMachineTemplate';
+
+const generateBuilders = (baseBuilder, groupIndex, keys, values) => {
+  const newComputedBuilders: { [k: string]: VMBuilder } = {};
+  Combinatorics.cartesianProduct(...values)
+    .toArray()
+    .forEach((additionalBuilderValues, additionalBuilderId) => {
+      const additionalBuilderData = additionalBuilderValues.reduce((acc, value, keyIdx) => {
+        acc[keys[keyIdx]] = value;
+        return acc;
+      }, {} as VMBuilderData);
+      const additionalBuilder = new VMBuilder().setData(additionalBuilderData);
+      newComputedBuilders[`config ${groupIndex}-${additionalBuilderId}`] = new VMBuilder(
+        baseBuilder,
+      )
+        .setBuilder(additionalBuilder)
+        .generateName(`${additionalBuilderId}`);
+    });
+  return newComputedBuilders;
+};
+
+export const computedVMBuilders = deepFreeze(
+  [
+    {
+      baseBuilder: new VMBuilder().setProvisionSource(provisionSources[ProvisionSource.CONTAINER]), // TODO Rest should be already preset
+      config: {
+        flavor: [{ flavor: Flavor.TINY }, { flavor: Flavor.SMALL }],
+        workload: [Workload.DESKTOP, Workload.SERVER],
+        os: [OperatingSystem.RHEL7, OperatingSystem.WINDOWS_10],
+      } as VMBuilderDataGenerationConfig,
+    },
+  ].reduce((_, { baseBuilder, config }, idx) => {
+    const configKeys = Object.keys(config);
+    const configValueArrays = configKeys.map((key) => config[key]);
+    return generateBuilders(baseBuilder, idx, configKeys, configValueArrays);
+  }, {}),
+);
+
+export const basicVM = new VMBuilder()
```

Comment:
suppose we want to use two basicVM builders in one test, then these two builders would be the same instance and would mutate themselves. 

I know it is always passed to the new VMBuilder constructor so this will never occur. But somebody might want to use  the basicVM without this knowledge in the future and it could cause issues

We have few options here:
 - make the data inside this VMBuilder immutable (frozen) - we could add freeze method on it
 - just use immutable VMBuilderData here instead
 - use a function which returns new instance of a VMBuilder

Reply:
> - use a function which returns new instance of a VMBuilder

updated